Libraries/libbom/Tools/lsbom.cpp

```diff
@@ -114,67 +114,91 @@ Options::
 {
 }
 
+static ext::optional<std::vector<Options::PrintItem>>
+ParsePrintFormat(std::string const &print)
+{
+    auto format = std::vector<Options::PrintItem>();
+
+    for (char c : print) {
+        switch (c) {
```

Comment:
Yah, although it'd be a few more lines of code and a little slower. :( `std::unordered_map` requires allocation and can't be `constexpr`, so it'd have to allocate memory at least once. Would also take an extra line to search in the map than just print out. A better language would make this easy...